passes/changedlines/git_api.go

```diff
@@ -0,0 +1,168 @@
+package changedlines
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+)
+
+// PRFile represents a file in a GitHub PR
+type PRFile struct {
```

Comment:
how do you think of defining an interface for loading git changes, and then capsulating this file to a object-like struct implemented the interface which is to load github API, and also for `git_local.go` is another implementation. then we can better organize the functions.